hostinfo/passwdstatus.lua

```diff
@@ -0,0 +1,113 @@
+--[[
+Copyright 2014 Rackspace
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--]]
+local HostInfo = require('./base').HostInfo
+
+local forEachLimit = require('async').forEachLimit
+local fs = require('fs')
+local os = require('os')
+local spawn = require('childprocess').spawn
+local table = require('table')
+
+--[[ Passwordstatus Variables ]]--
+local Info = HostInfo:extend()
+function Info:initialize()
+  HostInfo.initialize(self)
+end
+
+function Info:run(callback)
+  if os.type() ~= 'Linux' then
+    self._error = 'Unsupported OS for passwdstatus'
+    callback()
+    return
+  end
+
+  function collectData(users)
+    --[[ Asynchronous gather user passwd info ]]--
+    local obj = {}
+
+    forEachLimit(users, 5, function(user, callback)
+      local child
+      child = spawn('passwd', {'-S', user}, {})
```

Comment:
execFile *may* be useful there.  It lacks the stream end functionality which ensures the all the stream is read as opposed to assuming the stream is fully read on proc exit.  Fix it or create a new version like this?  https://gist.github.com/rjemanuele/4ac7473464b74b9a721d